src/v/kafka/tests/fetch_test.cc

```diff
@@ -281,3 +281,242 @@ FIXTURE_TEST(fetch_one, redpanda_thread_fixture) {
+
+    int i = 0;
+
+    for (auto it = response.begin(); it != response.end(); ++it) {
+        if (i < 3) {
+            BOOST_REQUIRE_EQUAL(it->partition->name(), "tp-1");
+            BOOST_REQUIRE_EQUAL(it->partition_response->id(), i);
+
+        } else if (i == 3) {
+            BOOST_REQUIRE_EQUAL(it->partition->name(), "tp-2");
+            BOOST_REQUIRE_EQUAL(it->partition_response->id(), 0);
+        } else {
+            BOOST_REQUIRE_EQUAL(it->partition->name(), "tp-3");
+            BOOST_REQUIRE_EQUAL(it->partition_response->id(), i - 4);
+        }
+        ++i;
+    }
+};
+
+FIXTURE_TEST(fetch_empty, redpanda_thread_fixture) {
+    // create a topic partition with some data
+    model::topic topic("foo");
+    model::partition_id pid(0);
+    model::offset offset(0);
+    auto ntp = make_default_ntp(topic, pid);
+    auto log_config = make_default_config();
+    wait_for_controller_leadership().get0();
+    add_topic(model::topic_namespace_view(ntp)).get();
+
+    tests::cooperative_spin_wait_with_timeout(10s, [this, ntp] {
+        auto shard = app.shard_table.local().shard_for(ntp);
+        if (!shard) {
+            return ss::make_ready_future<bool>(false);
+        }
+        return app.partition_manager.invoke_on(
+          *shard, [ntp](cluster::partition_manager& mgr) {
+              auto partition = mgr.get(ntp);
+              return partition
+                     && partition->committed_offset() >= model::offset(0);
+          });
+    }).get();
+
+    kafka::fetch_request no_topics;
+    no_topics.max_bytes = std::numeric_limits<int32_t>::max();
+    no_topics.min_bytes = 1;
+    no_topics.max_wait_time = std::chrono::milliseconds(1000);
+
+    auto client = make_kafka_client().get0();
+    client.connect().get();
+    auto resp_1 = client.dispatch(no_topics, kafka::api_version(6)).get0();
+
+    BOOST_REQUIRE(resp_1.partitions.empty());
+
+    kafka::fetch_request no_partitions;
+    no_partitions.max_bytes = std::numeric_limits<int32_t>::max();
+    no_partitions.min_bytes = 1;
+    no_partitions.max_wait_time = std::chrono::milliseconds(1000);
+    no_partitions.topics = {{.name = topic, .partitions = {}}};
+
+    auto resp_2 = client.dispatch(no_topics, kafka::api_version(6)).get0();
+    client.stop().then([&client] { client.shutdown(); }).get();
+
+    BOOST_REQUIRE(resp_2.partitions.empty());
+}
+
+
+FIXTURE_TEST(fetch_multi_partitions_debounce, redpanda_thread_fixture) {
+    // create a topic partition with some data
+    model::topic topic("foo");
+    model::partition_id pid(0);
+    model::offset offset(0);
+
+    wait_for_controller_leadership().get0();
+
+    add_topic(model::topic_namespace(model::ns("kafka"), topic), 6).get();
+
+    for (int i = 0; i < 6; ++i) {
+        auto ntp = make_default_ntp(topic, model::partition_id(i));
+        auto shard = app.shard_table.local().shard_for(ntp);
+        tests::cooperative_spin_wait_with_timeout(10s, [this, shard, ntp] {
```

Comment:
I feel like this is a really common thing to do, make some ntps , create a topic,  wait for controller leadership, then wait until the partition has data. Maybe some of this belongs in fixture.h and can be reused?